libs/src/api/serviceProvider.js

```diff
@@ -40,12 +40,22 @@ class ServiceProvider extends Base {
     }
 
     // Time requests and get version info
-    const timings = await Utils.timeRequests(
-      creatorNodes.map(node => ({
-        id: node.endpoint,
-        url: `${node.endpoint}/version`
-      }))
-    )
+    let timings = (await Promise.all(
```

Comment:
this is a really useful function, if we can make it generic this will be useful for other SP selection too - seems like it almost is already? like it can just accept a list and return all the timings